sycl/include/sycl/ext/intel/math/imf_bf16.hpp

```diff
@@ -0,0 +1,232 @@
+//==-------------------- imf_bf16.hpp - bfloat16 utils ---------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// C++ APIs for bfloat16 util functions.
+//===----------------------------------------------------------------------===//
+
+#pragma once
+#include <sycl/ext/oneapi/bfloat16.hpp>
+#include <type_traits>
+
+extern "C" {
+float __imf_bfloat162float(uint16_t);
+uint16_t __imf_float2bfloat16(float);
+uint16_t __imf_float2bfloat16_rd(float);
+uint16_t __imf_float2bfloat16_rn(float);
+uint16_t __imf_float2bfloat16_ru(float);
+uint16_t __imf_float2bfloat16_rz(float);
+};
+
+namespace sycl {
+__SYCL_INLINE_VER_NAMESPACE(_V1) {
+namespace ext {
+namespace intel {
+namespace math {
+
+// Need to ensure that sycl bfloat16 defined in bfloat16.hpp is compatible
+// with uint16_t in layout.
+#if __cplusplus >= 201703L
+static_assert(sizeof(sycl::ext::oneapi::bfloat16) == sizeof(uint16_t),
+              "sycl bfloat16 is not compatible with uint16_t.");
+
+float bfloat162float(sycl::ext::oneapi::bfloat16 b) {
+  return __imf_bfloat162float(__builtin_bit_cast(uint16_t, b));
+}
+
+sycl::ext::oneapi::bfloat16 float2bfloat16(float f) {
+  return __builtin_bit_cast(sycl::ext::oneapi::bfloat16,
```

Comment:
I think it might be recommended to use `oneapi::detail::bitsToBfloat16` here and in other places instead of `__builtin_bit_cast` ?

Reply:
Hi, @JackAKirk 
I didn't realize sycl bfloat16 has already had bitsToBfloat16, will update the code if we move the generic implementation to bfloat16_math.hpp
Thanks very much.